### PR TITLE
[DOC] Added MOIRAIforecaster to the API docs, create new section on pretrained FM

### LINKED ISSUE #7217: [DOC] `MOIRAIForecaster` is not showing up in the API documentation
#### Describe the issue linked to the documentation

https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.forecasting.moirai_forecaster.MOIRAIForecaster.html

#### Suggest a potential alternative/fix

Add the forecaster to the rst file

### PR DESCRIPTION
**Pull request by Dehelaan**
**Reference Issue**
Fixes #7217

It Adds the `MOIRAIforecaster` to the rst files in docs\source\api_reference\forecasting.rst

- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md)

### CODE REVIEW

File: docs/source/api_reference/forecasting.rst [hunk 1]
```diff
@@ -589,6 +589,17 @@ Generic framework adapters that expose other frameworks in the ``sktime`` interf
 
     HCrystalBallAdapter
 
+MOIRAI Forecasters
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply docs/source/api_reference/forecasting.rst:592]: hm, but it is not a reducer - it should be with the deep learning models. Is there a section on foundation models?

[the remark this answers is not in the document]
Comment (Dehelaan) [reply docs/source/api_reference/forecasting.rst]: Thank you for pointing it out. I have tried to resolve the issue. Also, YES there is a section (Deep learning based forecasters) where I added it.

**Review by fkiraly — APPROVED** [review]
Nice! I also added a new section for pretrained models

**Comment by Dehelaan** [comment]
@fnhirwa

**Review by fnhirwa — APPROVED** [review]
Looks good to me

**Pull request merged**